### PR TITLE
Jon/aave/prevent-excess-action-amount

### PR DESCRIPTION
**Pull request by Jon-edge**
- Cap tile display amounts to handle negative inputs
- Prevent over withdraws/repays with capped values

#### PR Dependencies

 none

#### PR Requirements

If you have made **any** visual changes to the GUI. Make sure you have:
- [ ] Tested on iOS device
- [ ] Tested on Android device
- [ ] Tested on small-screen device (iPod Touch)
- [ ] Tested on large-screen device (tablet)

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1203169636825094

### CODE REVIEW

File: src/components/scenes/Loans/LoanManageScene.tsx [hunk 1]
```diff
@@ -158,17 +162,41 @@ export const LoanManageScene = (props: Props) => {
...
+    if (actionOpType === 'loan-withdraw') {
+      availableNativeAmount = collaterals.find(collateral => collateral.tokenId === selectedAsset.tokenId)?.nativeAmount
+    } else if (actionOpType === 'loan-repay') {
+      availableNativeAmount = debts.find(debt => debt.tokenId === selectedAsset.tokenId)?.nativeAmount
+    } else {
+      availableNativeAmount = undefined // Use uncapped actionNativeAmount
+    }
+
+    const cappedActionNativeAmount =
+      availableNativeAmount != null && (availableNativeAmount == null || gt(actionNativeAmount, availableNativeAmount))
```
Comment (samholmes) [inline src/components/scenes/Loans/LoanManageScene.tsx:188]: Isn't `availableNativeAmount != null && (availableNativeAmount == null...` redundant logic?

**Review by samholmes — CHANGES_REQUESTED** [review]
Shouldn't we be checking wallet balance and doing a min of the wallet balance for debt repayment?

File: src/components/tiles/LtvRatioTile.tsx [hunk 2]
```diff
@@ -47,5 +47,7 @@ export const LtvRatioTile = (props: {
 
   const futureValue = zeroString(totalCollateralFiatValue) ? '0' : div(totalDebtFiatValue, totalCollateralFiatValue, 2)
 
-  return <PercentageChangeArrowTile title={s.strings.loan_loan_to_value_ratio} currentValue={currentValue} futureValue={futureValue} />
+  return (
+    <PercentageChangeArrowTile title={s.strings.loan_loan_to_value_ratio} currentValue={currentValue} futureValue={lt(futureValue, '0') ? '0' : futureValue} />
```
[the remark this answers is not in the document]
Comment (Jon-edge) [reply src/components/tiles/LtvRatioTile.tsx:51]: I'll change it, but this is one of those situations where I'd prefer leaving it longer for better readability.

File: src/components/tiles/TotalDebtCollateralTile.tsx [hunk 3]
```diff
@@ -1,12 +1,14 @@
...
 import { useTotalFiatAmount } from '../../util/borrowUtils'
+import { zeroString } from '../../util/utils'
 import { FiatAmountTile } from './FiatAmountTile'
 
 export const TotalDebtCollateralTile = (props: { title: string; wallet: EdgeCurrencyWallet; debtsOrCollaterals: any[] }) => {
   const { title, wallet, debtsOrCollaterals } = props
   const totalFiatAmount = useTotalFiatAmount(wallet, debtsOrCollaterals)
 
-  return <FiatAmountTile title={title} fiatAmount={totalFiatAmount} wallet={wallet} />
+  return <FiatAmountTile title={title} fiatAmount={!zeroString(totalFiatAmount) && gt(totalFiatAmount, '0') ? totalFiatAmount : '0'} wallet={wallet} />
```
[the remark this answers is not in the document]
Comment (Jon-edge) [reply src/components/tiles/TotalDebtCollateralTile.tsx:13]: I'll change it, but this is one of those situations where I'd prefer leaving it longer for better readability.

File: src/components/scenes/Loans/LoanManageScene.tsx (hunk 1 above)
Comment (Jon-edge) [reply src/components/scenes/Loans/LoanManageScene.tsx:188]: Yeah, it is, but it was to get TS to stop complaining about `cappedActionNativeAmount` being potentially undefined.

I'll rework the logic here so it makes more sense

**Comment by Jon-edge** [comment]
No, we should roll that into the max button task. It's a little more nuanced than that because the fund source isn't necessarily going to be the same as the debt asset.

File: src/components/tiles/LtvRatioTile.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (samholmes) [reply src/components/tiles/LtvRatioTile.tsx]: Using `max()` is a very common in mathematics for setting a lower boundary. The name space for min/max is unpacked from some academic work I guess. I agree it takes a little getting used to, but it's the correct function for this.

**Review by samholmes — APPROVED** [review]

**Pull request merged**